language/move-lang/src/parser/syntax.rs

```diff
@@ -738,16 +704,32 @@ fn parse_term<'input>(tokens: &mut Lexer<'input>) -> Result<Exp, Error> {
     Ok(spanned(tokens.file_name(), start_loc, end_loc, term))
 }
 
-// Parse the subset of expression terms for pack and call operations, as well as for generic names
-// (in specifications only).
-// This is a helper function for parse_term.
-fn parse_pack_or_call_or_generic_name<'input>(tokens: &mut Lexer<'input>) -> Result<Exp_, Error> {
+// Parse a pack, call, or other reference to a name:
+//      NameExp =
+//          <ModuleAccess> <OptionalTypeArgs> "{" Comma<ExpField> "}"
+//          | <ModuleAccess> <OptionalTypeArgs> "(" Comma<Exp> ")"
+//          | <ModuleAccess> <OptionalTypeArgs>
```

Comment:
Want to bring back the `(spec only)` comment for this variant?

Reply:
I removed the comment because with this change it is not just for specs.

Reply:
I should clarify that: a generic name with type arguments is still only allowed for specs, but if the type args are not used, then this is the syntax to refer to local variables.